### PR TITLE
Add hexadecimal check (fix warnings)

### PR DESCRIPTION
**Pull request by nupplaphil**
Fixing https://github.com/friendica/friendica/issues/6917#issuecomment-475461338
Fixing https://github.com/friendica/friendica/issues/6917#issuecomment-492798238

### CODE REVIEW

File: mod/photos.php [hunk 1]
```diff
@@ -1027,6 +1027,10 @@ function photos_content(App $a)
 
 	// Display a single photo album
 	if ($datatype === 'album') {
+		// if $datum is not a valid hex, redirect to the default page
+		if (!Strings::isHex($datum)) {
+			$a->internalRedirect();
```
Comment (MrPetovan) [inline mod/photos.php]: Can you please redirect to `/photos` instead?

File: mod/parse_url.php [hunk 2]
```diff
@@ -25,10 +27,13 @@ function parse_url_content(App $a)
...
 
-	if (!empty($_GET['binurl'])) {
+	if (!empty($_GET['binurl']) && Strings::isHex($_GET['binurl'])) {
 		$url = trim(hex2bin($_GET['binurl']));
-	} else {
+	} elseif (!empty($_GET['url'])) {
 		$url = trim($_GET['url']);
+	// fallback in case no url is valid
+	} else {
+		$a->internalRedirect();
```
Comment (MrPetovan) [inline mod/parse_url.php]: Can you please redirect to `/parse_url` instead?

File: mod/photos.php (hunk 1 above)
Comment (nupplaphil) [reply mod/photos.php]: `/photos` without a parameter just returns the base-pagelayout with "no photos selected". That's why I thought it's better to redirect to the landing page

File: mod/parse_url.php (hunk 2 above)
Comment (nupplaphil) [reply mod/parse_url.php]: We would create an infinite loop in this case.

Comment (MrPetovan) [reply mod/parse_url.php]: Ha, nevermind, it should just return empty string instead. it is supposed to be a `rawContent` endpoint, never to be accessed directly with a browser, so a redirection still is unwarranted.

File: mod/photos.php (hunk 1 above)
Comment (MrPetovan) [reply mod/photos.php]: Indeed, what about `/photos/{nickname}` instead?

**Pull request merged**